chain/consensus/hierarchical/actors/sca/sca_apply.go

```diff
@@ -190,3 +195,23 @@ func noopWithStateTransaction(rt runtime.Runtime, msg types.Message, code exitco
 		noop(rt, &st, msg, code, err)
 	})
 }
+
+func recordCrossNetMsgExecution(rt runtime.Runtime, msg types.Message, code exitcode.ExitCode, msgType string) {
```

Comment:
Same here, this is executed by every peer in the network, you need to find a better way to track these messages without coupling new code in the actor. 

A good way of doing so is through the event listener. You can check `TopDownMsg` and `BottomUpMsg` variables and see if there are new messages to be executed.